backend/agentd/session.go

```diff
@@ -495,8 +495,13 @@ func (s *Session) Start() (err error) {
 		s.mu.Unlock()
 	}
 
+	s.mu.Lock()
+	subs := make([]string, len(s.cfg.Subscriptions))
+	copy(subs, s.cfg.Subscriptions)
+	s.mu.Unlock()
+
```

Comment:
Premature optimization opportunity: should we make `mu` a `sync.RWMutex` instead of a `sync.Mutex`? This block only reads from `s.cfg.Subscriptions`.

Reply:
I believe the contention is not low enough to make a difference. There are only two goroutines involved.